### PR TITLE
ci: uncomment docs build

### PR DESCRIPTION
**Pull request by Pranavchiku**

### CODE REVIEW

**Comment by Pranavchiku** [comment]
Documentation failure:

```consotle
Test project /home/runner/work/lfortran/lfortran
    Start 1: test_cwrapper
1/3 Test #1: test_cwrapper ....................   Passed    0.00 sec
    Start 2: test_stacktrace
2/3 Test #2: test_stacktrace ..................   Passed    0.00 sec
    Start 3: test_lfortran
3/3 Test #3: test_lfortran ....................***Failed    0.30 sec
Errors while running CTest
<string>:5:9: error: expected instruction opcode
    %1 =x alloca i64
        ^
<string>:5:9: error: expected instruction opcode
    %1 =x alloca i64
        ^
<string>:5:25: error: use of undefined value '@count'
    %1 = load i64, i64* @count
                        ^
[doctest] doctest version is "2.4.8"
[doctest] run with "--help" for options
===============================================================================
/home/runner/work/lfortran/lfortran/src/lfortran/tests/test_llvm.cpp:889:
TEST CASE:  FortranEvaluator 7
/home/runner/work/lfortran/lfortran/src/lfortran/tests/test_llvm.cpp:901: ERROR: CHECK( r.result.i32 == 5 ) is NOT correct!
  values: CHECK( 0 == 5 )
===============================================================================
/home/runner/work/lfortran/lfortran/src/lfortran/tests/test_llvm.cpp:904:
TEST CASE:  FortranEvaluator 8
/home/runner/work/lfortran/lfortran/src/lfortran/tests/test_llvm.cpp:916: ERROR: CHECK( r.result.f32 == 3.5 ) is NOT correct!
  values: CHECK( 0.0f == 3.5 )
===============================================================================
/home/runner/work/lfortran/lfortran/src/lfortran/tests/test_llvm.cpp:919:
TEST CASE:  FortranEvaluator 8 double
/home/runner/work/lfortran/lfortran/src/lfortran/tests/test_llvm.cpp:931: ERROR: CHECK( r.result.f64 == 3.5 ) is NOT correct!
  values: CHECK( 0.0 == 3.5 )
0    0    0    0    0    0    0    0    0    0
5    5    5
src/lfortran/tests/ir.ll: error: Could not open input file: No such file or directory
src/lfortran/tests/ir.ll: error: Could not open input file: No such file or directory
===============================================================================
[doctest] test cases:  95 |  92 passed | 3 failed | 0 skipped
[doctest] assertions: 672 | 669 passed | 3 failed |
[doctest] Status: FAILURE!
67% tests passed, 1 tests failed out of 3
Total Test time (real) =   0.35 sec
The following tests FAILED:
	  3 - test_lfortran (Failed)
```

**Comment by Pranavchiku** [comment]
For

```fortran
program main
integer :: i = 5
print *, i
end program
```


Correct ASR:

```clojure
(TranslationUnit
    (SymbolTable
        1
        {
            main:
                (Program
                    (SymbolTable
                        2
                        {
                            i:
                                (Variable
                                    2
                                    i
                                    []
                                    Local
                                    (IntegerConstant 5 (Integer 4) Decimal)
                                    (IntegerConstant 5 (Integer 4) Decimal)
                                    Save
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main
                    []
                    [(Print
                        [(StringFormat
                            ()
                            [(Var 2 i)]
                            FormatFortran
                            (Character -1 0 ())
                            ()
                        )]
                        ()
                        ()
                    )]
                )
        })
    []
)
```

The problem was that simplifier created an Assignment node for Save value:

```clojure
(TranslationUnit
    (SymbolTable
        1
        {
            main:
                (Program
                    (SymbolTable
                        2
                        {
                            i:
                                (Variable
                                    2
                                    i
                                    []
                                    Local
                                    ()
                                    ()
                                    Save
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main
                    []
                    [(Assignment
                        (Var 2 i)
                        (IntegerConstant 5 (Integer 4) Decimal)
                        ()
                    )
                    (Print
                        (StringFormat
                            ()
                            [(Var 2 i)]
                            FormatFortran
                            (Character -1 0 () PointerString)
                            ()
                        )
                    )]
                )
        })
    []
)
```

Which won't hurt when we compile one file but in multi file compilation it would hurt.

**Comment by Pranavchiku** [comment]
CI shall now work

File: src/libasr/pass/simplifier.cpp [hunk 1]
```diff
@@ -2319,6 +2324,10 @@ class VerifySimplifierASROutput:
             !(check_if_ASR_owner_is_module(x.m_parent_symtab->asr_owner)) &&
             !(check_if_ASR_owner_is_enum(x.m_parent_symtab->asr_owner)) &&
             !(check_if_ASR_owner_is_struct(x.m_parent_symtab->asr_owner)) &&
+            !(x.m_storage == ASR::storage_typeType::Save && x.m_symbolic_value &&
```
Comment (gxyd) [inline src/libasr/pass/simplifier.cpp:2327]: Having an integration test for this would be nice. Maybe in *main*?

Comment (gxyd) [reply src/libasr/pass/simplifier.cpp:2327]: Ah I see, I don't think there is any need. It will be reflected in reference updates I think

**Review by certik — APPROVED** [review]

**Pull request merged**